picard/formats/vorbis.py

```diff
@@ -69,6 +69,27 @@ def is_valid_key(key):
     return INVALID_CHARS.search(key) is None
 
 
+def flac_sort_pics_after_tags(metadata_blocks):
+    # First remember all picture blocks that are located before the tag block.
+    tagblock = None
+    picblocks = []
+    for b in metadata_blocks:
+        if b.code == mutagen.flac.VCFLACDict.code:
+            tagblock = b
+            break
+        elif b.code == mutagen.flac.Picture.code:
+            picblocks.append(b)
+    else:
+        # No tags found, nothing to sort
+        return
+
+    # Now move those picture block after the tag block, maintaining their order.
```

Comment:
> maintaining their order ?

Perhaps I'm missing something here.
List of pictures is built from previous loop, using `append()`, but `insert(index, value)` will actually revere the order:

```python
>>> l = ['a', 'b', 'c']
>>> l
['a', 'b', 'c']
>>> for p in ['pic1', 'pic2', 'pic3']:
...     l.insert(1, p)
... 
>>> l
['a', 'pic3', 'pic2', 'pic1', 'b', 'c']
```

Reply:
Ha, my first thinking was also that I would need to insert at tagindex + 1 and then increment. But turns out the insert position is stable at tagindex, because one by one the blocks are removedat the start and then inserted again. The removal automatically moves all following elements, making room for the insertion.

Reply:
Just to demonstrate what I mean with your example:

```python
>>> l = ['pic1', 'pic2', 'pic3', 'a', 'b', 'c']
>>> l
['pic1', 'pic2', 'pic3', 'a', 'b', 'c']
>>> for p in ['pic1', 'pic2', 'pic3']:
...     l.remove(p)
...     l.insert(3, p)
...
>>> l
['a', 'pic1', 'pic2', 'pic3', 'b', 'c']

```

Reply:
I got it, missed this remove part ;)